docs/standard/commandline/snippets/get-started-tutorial/csharp/Stage3/Program.cs

```diff
@@ -0,0 +1,147 @@
+﻿// <all>
+using System.CommandLine;
+
+namespace scl;
+
+class Program
+{
+    static async Task<int> Main(string[] args)
+    {
+        // <fileoption>
+        var fileOption = new Option<FileInfo?>(
+            name: "--file",
+            description: "An option whose argument is parsed as a FileInfo",
+            isDefault: true,
+            parseArgument: result =>
+            {
+                if (result.Tokens.Count == 0)
+                {
+                    return new FileInfo("sampleQuotes.txt");
+
+                }
+                var filePath = result.Tokens.Single().Value;
+                if (!File.Exists(filePath))
+                {
+                    result.ErrorMessage = "File does not exist";
```

Comment:
Consider:

// This allows customization of the error behavior, like a custom message. Without this, missing files will still be reported.